product-service/handlers/postProduct.ts

```diff
@@ -0,0 +1,96 @@
+import { APIGatewayEvent, APIGatewayProxyResult} from 'aws-lambda';
+import 'source-map-support/register';
+import { dbOptions } from '../db/db-options';
+import { Client } from 'pg';
+import middy from '@middy/core';
+import validator from '@middy/validator';
+import httpErrorHandler from '@middy/http-error-handler';
+import jsonBodyParser from '@middy/http-json-body-parser';
+
+const postProduct = middy(async (event: APIGatewayEvent): Promise<APIGatewayProxyResult> => {
+  const { httpMethod, path, body } = event;
+  console.log(JSON.stringify({ httpMethod, path, body }, null, 2))
+
+  const client = new Client(dbOptions);
+  await client.connect();
+
+  try {
+    const { title, description, price, image, count }: any = body;
+
+    await client.query('BEGIN');
+    const queryTextProducts = 'INSERT into products (title, description, price, image) VALUES ($1, $2, $3, $4) RETURNING id';
+    const queryValuesProducts = [title, description, price, image];
+    const responseProducts = await client.query(queryTextProducts, queryValuesProducts)
+
+    const queryTextStocks = 'INSERT into stocks (count, product_id) VALUES ( $1, $2)';
+    const productId = responseProducts.rows[0].id;
+    const queryValuesStocks = [count, productId];
+    await client.query(queryTextStocks, queryValuesStocks);
+
+    await client.query('COMMIT');
+
+    const { rows: product } = await client.query(`
+    SELECT p.*, s.count FROM products p LEFT JOIN stocks s ON p.id = s.product_id WHERE p.id = $1`,
+    [productId]);
+  
+    
+    if (!product) {
+      return {
+        statusCode: 403,

```

Comment:
Please return 400 status code. 403 is used when user is not authorized to perform this action.